syscall_linux.go

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build !cgo && (amd64 || arm64)
+
+package purego
+
+import (
+	"math"
+	"unsafe"
+)
+
+var syscall9XABI0 uintptr
+
+type syscall9Args struct {
+	fn, a1, a2, a3, a4, a5, a6, a7, a8, a9 uintptr
+	f1, f2, f3, f4, f5, f6, f7, f8         float64
+	r1, r2, err                            uintptr
+}
+
+//go:nosplit
+func syscall_syscall9X(fn, a1, a2, a3, a4, a5, a6, a7, a8, a9 uintptr) (r1, r2, err uintptr) {
+	args := syscall9Args{fn, a1, a2, a3, a4, a5, a6, a7, a8, a9,
+		math.Float64frombits(uint64(a1)), math.Float64frombits(uint64(a2)), math.Float64frombits(uint64(a3)),
+		math.Float64frombits(uint64(a4)), math.Float64frombits(uint64(a5)), math.Float64frombits(uint64(a6)),
+		math.Float64frombits(uint64(a7)), math.Float64frombits(uint64(a8)),
+		r1, r2, err}
+	runtime_cgocall(syscall9XABI0, unsafe.Pointer(&args))
+	return args.r1, args.r2, args.err
+}
+
+func NewCallback(_ interface{}) uintptr {
+	panic("purego: NewCallback not supported")
+}
```

Comment:
Do we need the definition of `NewCallback` for Linux, if this doesn't work?

Reply:
I don't think we need it for Oto, and OpenGL doesn't need it nor does Linux gamepad. It'll probably be needed once we try removing glfw but that will be a while before that happens.

Reply:
OK. I prefer removing this than leaving this with panic.

Reply:
It can't be removed since it will fail to compile in RegisterFunc which tries to convert func arguments to C functions.

Reply:
Does this mean everything with RegisterFunc doesn't work on Linux?

Reply:
Yes, callbacks won't work and floats won't work on Cgo but do without Cgo. Everything else that works on Darwin works on Linux

Reply:
OK I understand.

Reply:
Can we have comments somewhere about what works and what not so that purego users can understand?

Reply:
I added the documentation to RegisterFunc!

Reply:
Thanks!